source/common/http/conn_manager_impl.cc

```diff
@@ -1012,6 +1027,13 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
     }
   }
 
+  // If we're encoding a headers only response, then the HTTP/1 codec will reset the request
+  // when we encode the headers, so set this to prevent us from attempting to reset it again
+  // in doEndStream.
+  if (encoding_headers_only_) {
+    state_.local_complete_ = true;
```

Comment:
Thanks I will spend some time with this today.

Reply:
Sorry I see this now. Yes this makes sense. We need to implicitly say that we are ending the local stream. Yay for integration tests! I wonder if we should actually set this as soon as a filter indicates that it is headers only? This is basically already what is set inside `commonEncodePrefix`? Not sure if it matters but throwing that out there as something to analyze.

Reply:
I don't think it matters since `local_complete_` seems to only be read in `doEndEncode`, but I'll move it closer because I think it makes more sense conceptually